src/auto-aspect-executor.ts

```diff
@@ -45,33 +48,26 @@ export class AutoAspectExecutor implements OnModuleInit {
             originalFn: any;
             metadata?: unknown;
             aopSymbol: symbol;
-          }[] = this.reflector.get(metadataKey, instance[methodName]);
+          }[] = this.reflector.get(metadataKey, target[methodName]);
           if (!metadataList) {
             return;
           }
 
-          // TODO: Support request scope providers
-          if (!wrapper.isDependencyTreeStatic()) {
-            console.warn(
-              `[${
-                wrapper.instance?.constructor?.name || 'Unknown'
-              }] "@tossteam/nestjs-aop" does not yet support request scope providers`,
-            );
-            return;
-          }
-
           for (const { originalFn, metadata, aopSymbol } of metadataList) {
-            const wrappedMethod = lazyDecorator.wrap({
-              instance,
-              methodName,
-              method: originalFn.bind(instance),
-              metadata,
+            const proxy = new Proxy(target[methodName], {
+              apply: (_, thisArg, args) => {
+                const wrappedMethod = lazyDecorator.wrap({
+                  instance: thisArg,
+                  methodName,
+                  method: originalFn.bind(thisArg),
+                  metadata,
+                });
+                return Reflect.apply(wrappedMethod, lazyDecorator, args);
+              },
```

Comment:
First of all, thank you for your contribution!

If the wrap method is executed inside the application function, it is likely that the wrap will be called every call.
Is this necessary because "thisArg" changes every time in the request scope?

Reply:
Yes. Current approach is applying lazy decorators at `onModuleInit` time, and in that time request scoped wrappers don't have their instance, so instance can't bind at applying time. 

I considered another approach that apply a global interceptor and the interceptor apply lazy decorators to request scoped wrappers. In this way it is not necessary that you'd comment. But that approach can't apply lazy decorators when getting instance by `moduleRef.resolve()`.

Any suggestion for better way would be appreciated, thanks.

Reply:
How about caching the wrapped methods created using WeakMap?

Reply:
+ If only static instances are created and cached in the onModuleInit stage, there will be no delay due to lazy generation.